metadrive/engine/base_engine.py

```diff
@@ -324,7 +326,8 @@ def step(self, step_num: int = 1) -> None:
                     manager.step()
             self.step_physics_world()
             # the recording should happen after step physics world
-            if "record_manager" in self.managers:
+            if "record_manager" in self.managers and i < step_num - 1:
+                # last recording should be finished in after_step, as some objects may be created in after_step
```

Comment:
Can't understand this comment. Try to explain better.

Reply:
We repeat run simulator ```decision_num``` times, and record after each running. While the last time is actually finished when all managers finish the ```after_step()``` function. So the recording for the last time should be done after that. Like in ```PGTrafficManager``` we may create new vehicles in ```after_step()```. These new cars' states can be recorded only if we run ```record_managers.step()``` after their creation. So the recording of the last running is  in ```record_manager.after_step()```